lib/ansible/template/__init__.py

```diff
@@ -98,13 +98,36 @@ def generate_ansible_template_vars(path, fullpath=None, dest_path=None):
     else:
         temp_vars['template_fullpath'] = fullpath
 
+    safe = True
+    morsel = ''
+
+    t_file = temp_vars['template_path'].replace('%', '%%')
+    t_host = temp_vars['template_host']
+    t_uid = temp_vars['template_uid']
+
+    if is_possibly_template(t_file, j2env):
+        safe = False
+        morsel += f' template file name({t_file})'
+
+    if is_possibly_template(t_host, j2env):
+        safe = False
+        morsel += f' host({t_host})'
+
+    if is_possibly_template(t_uid, j2env):
+        safe = False
+        morsel += f' uid({t_uid})'
+
     managed_default = C.DEFAULT_MANAGED_STR
     managed_str = managed_default.format(
-        host=temp_vars['template_host'],
-        uid=temp_vars['template_uid'],
-        file=temp_vars['template_path'].replace('%', '%%'),
+        host=t_host,
+        uid=t_uid,
+        file=t_file,
```

Comment:
Convert the placeholders to templates, then all the other changes to this file can be discarded, except for the removal of `to_unsafe_text`:

```suggestion
        host="{{ template_host }}",
        uid="{{ template_uid }}",
        file="{{ template_path }}"
```

Reply:
but that is what we want to avoid ...., the part we want to enable templating is for the original string

Reply:
That's what this does. The vars in `temp_vars`, if they're at risk of containing templates, should be passed through `to_unsafe_text` to prevent them from being templated.

Reply:
but that makes the whole string unsafe, which prevents the other parts from being templated